lib/jsonapi/serializer.ex

```diff
@@ -144,29 +152,31 @@ defmodule JSONAPI.Serializer do
     |> Enum.uniq
   end
 
-  # This makes a mapping between includes from the query parser and includes in the view.
-  defp get_includes(view, nil), do: view.relationships()
-  defp get_includes(view, []), do: view.relationships()
-  defp get_includes(view, query_includes) when is_list(query_includes) do
-    base = view.relationships()
-    Enum.reduce(query_includes, [], &(handle_include(base, &1, &2)))
-  end
-  defp get_includes(view, include) do
-    base = view.relationships()
-    Keyword.get(base, include)
+  defp get_includes(view, query_includes) do
+    get_default_includes(view) ++ get_query_includes(view, query_includes)
+    |> Enum.uniq
   end
 
-  defp handle_include(base, {parent, child}, acc) do
-    view = Keyword.get(base, parent)
-    acc = Keyword.put(acc, parent, view)
-    handle_include(view, child, acc)
-  end
-  defp handle_include({base, :include}, child, acc) do
-    handle_include(base.relationships(), child, acc)
+  defp get_default_includes(view) do
+    rels = view.relationships()
+    default_includes = rels |> Enum.filter(fn {k, v} ->
+      case v do
+        {_, :include} -> true
+        _ -> false
+      end
+    end)
   end
-  defp handle_include(base, child, acc) do
-    view = if is_list(base), do: Keyword.get(base, child), else: base
-    Keyword.put(acc, child, view)
+
+  defp get_query_includes(view, query_includes) do
+    rels = view.relationships()
+    Enum.map(query_includes, fn include ->
```

Comment:
I think there is an opportunity to refactor this.  We could probably move to `Enum.reduce/3` and we can eliminate the `case` by matching on the function signature instead.